loader/loader.go

```diff
@@ -47,114 +54,188 @@ var (
 		{"cdnjs", cdnjs, regexp.MustCompile(`^cdnjs.com/libraries/([^/]+)(?:/([(\d\.)]+-?[^/]*))?(?:/(.*))?$`)},
 		{"github", github, regexp.MustCompile(`^github.com/([^/]+)/([^/]+)/(.*)$`)},
 	}
-	invalidScriptErrMsg = `The file "%[1]s" couldn't be found on local disk, ` +
-		`and trying to retrieve it from https://%[1]s failed as well. Make ` +
-		`sure that you've specified the right path to the file. If you're ` +
+	httpsSchemeCouldntBeLoadedMsg = `The moduleSpecifier "%s" couldn't be retrieved from` +
+		` the resolved url "%s". Error : "%s"`
+	fileSchemeCouldntBeLoadedMsg = `The moduleSpecifier "%s" couldn't be found on ` +
+		`local disk. Make sure that you've specified the right path to the file. If you're ` +
 		`running k6 using the Docker image make sure you have mounted the ` +
 		`local directory (-v /local/path/:/inside/docker/path) containing ` +
 		`your script and modules so that they're accessible by k6 from ` +
 		`inside of the container, see ` +
 		`https://docs.k6.io/v1.0/docs/modules#section-using-local-modules-with-docker.`
+	errNoLoaderMatched = errors.New("no loader matched")
 )
 
-// Resolves a relative path to an absolute one.
-func Resolve(pwd, name string) string {
-	if name[0] == '.' {
-		return filepath.ToSlash(filepath.Join(pwd, name))
+// Resolve a relative path to an absolute one.
+func Resolve(pwd *url.URL, moduleSpecifier string) (*url.URL, error) {
+	if moduleSpecifier == "" {
+		return nil, errors.New("local or remote path required")
 	}
-	return name
-}
 
-// Returns the directory for the path.
-func Dir(name string) string {
-	if name == "-" {
-		return "/"
+	// we always want for the pwd to end in a slash, but filepath/path.Clean strips it so we readd
+	// it if it's missing
+	if pwd.Opaque != "" {
+		if !strings.HasSuffix(pwd.Opaque, "/") {
+			pwd.Opaque += "/"
+		}
+	} else if !strings.HasSuffix(pwd.Path, "/") {
+		pwd.Path += "/"
 	}
-	return filepath.Dir(name)
-}
 
-func Load(fs afero.Fs, pwd, name string) (*lib.SourceData, error) {
-	log.WithFields(log.Fields{"pwd": pwd, "name": name}).Debug("Loading...")
+	if moduleSpecifier[0] == '.' || moduleSpecifier[0] == '/' {
```

Comment:
This concerns me when it comes to absolute file paths on windows...

Reply:
How about now?